src/ServiceControl.Plugin.Nsb6.SagaAudit/CaptureSagaStateBehavior.cs

```diff
@@ -2,41 +2,43 @@
 {
     using System;
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using EndpointPlugin.Messages.SagaState;
     using NServiceBus;
     using NServiceBus.Pipeline;
-    using NServiceBus.Pipeline.Contexts;
-    using NServiceBus.Saga;
+    using NServiceBus.Routing;
     using NServiceBus.Sagas;
 
-    class CaptureSagaStateBehavior : IBehavior<IncomingContext>
+    class CaptureSagaStateBehavior : Behavior<IInvokeHandlerContext>
     {
-        Configure configure;
         SagaUpdatedMessage sagaAudit;
         ServiceControlBackend backend;
+        EndpointName endpointName;
+        readonly CaptureSagaStateSerializer serializer;
 
-        public CaptureSagaStateBehavior(Configure configure, ServiceControlBackend backend)
+        public CaptureSagaStateBehavior(EndpointName endpointName, CaptureSagaStateSerializer serializer, ServiceControlBackend backend)
         {
-            this.configure = configure;
+            this.endpointName = endpointName;
+            this.serializer = serializer;
             this.backend = backend;
         }
 
-        public void Invoke(IncomingContext context, Action next)
+        public override async Task Invoke(IInvokeHandlerContext context, Func<Task> next)
         {
             var saga = context.MessageHandler.Instance as Saga;
 
             if (saga == null)
             {
-                next();
-                return;
+                await next().ConfigureAwait(false);
```

Comment:
I will tackle this today. Thanks bob for the help!

On Wed, Mar 16, 2016 at 7:47 AM, Mike Minutillo <notifications@github.com>
wrote:

> In src/ServiceControl.Plugin.Nsb6.SagaAudit/CaptureSagaStateBehavior.cs
> <https://github.com/Particular/ServiceControl.Plugin.Nsb6.SagaAudit/pull/2#discussion_r56287717>
> :
>
> >          {
> >              var saga = context.MessageHandler.Instance as Saga;
> >
> >              if (saga == null)
> >              {
> > -                next();
> > -                return;
> > +                await next().ConfigureAwait(false);
>
> When the call to next() returns this will continue to the rest of the
> method. This is meant to be an early termination point but it's not
> currently working that way.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/ServiceControl.Plugin.Nsb6.SagaAudit/pull/2/files/6ef188210ec18800e8bbe5600e6868c95cc45568#r56287717>
>